net/dns/manager_linux.go

```diff
@@ -50,6 +51,15 @@ func NewOSConfigurator(logf logger.Logf, interfaceName string) (ret OSConfigurat
 	switch resolvOwner(bs) {
 	case "systemd-resolved":
 		dbg("rc", "resolved")
+		// Some systems, for reasons known only to them, have a
```

Comment:
I think it's because LXC takes its host resolv.conf and bind mounts it into the container. The host probably has systemd-resolved.

Maybe detect running-in-LXC instead? We have that code elsewhere in the tree.

Reply:
The format of the resolv.conf doesn't match anything written by systemd-resolved though. It's as if LXC had made up a config, and then for the lulz just added `# systemd-resolved` at the top of the file.

I'd rather not do something special for LXC, because when appropriately configured our existing logic will do the right thing. This is just the first time I've seen a resolv.conf that claims to belong to systemd-resolved, but was actually plopped down by something completely other with no explanation.

Reply:
I think I probably did this: my notes for a different LXC include installing Tailscale and experimenting with getting DNS settings to work. I had a problem with Proxmox overwriting /etc/resolv.conf.

If I edited /etc/resolv.conf in the wrong window and changed the Ubuntu host's resolver config, then that would explain why the LXC I was trying to work with didn't do what I expected.

Reply:
Hmm. So this may be too isolated an issue to warrant changing our code... That said, the additional check is harmless IMO, in that it won't start doing the wrong thing on configs that used to work, and may do better in a few more edge cases. On that basis, I think I'm going to merge.